lang/cs/Org.Apache.REEF.IMRU/OnREEF/ResultHandler/WriteResultHandler.cs

```diff
@@ -84,12 +89,31 @@ public void Dispose()
             if (_fileSystem.Exists(remoteUri))
             {
                 Exceptions.Throw(
-                    new Exception(string.Format("Output Uri: {0} already exists", remoteUri)), Logger);
+                    new Exception(string.Format("Output Uri: {0} already exists", remoteUri)),
+                    Logger);
             }
 
             _fileSystem.CopyFromLocal(_localFilename, remoteUri);
         }
 
+        /// <summary>
+        /// Delete local file
+        /// </summary>
+        public void Dispose()
+        {
+            if (_localFilename != null)
```

Comment:
Do we need double check for `null` here? Once inside the lock should be sufficient